src/main/java/hudson/plugins/analysis/core/PluginDescriptor.java

```diff
@@ -122,21 +128,6 @@ public PluginDescriptor(final Class<? extends Publisher> clazz) {
     }
 
     @Override
-    public Publisher newInstance(final StaplerRequest req, final JSONObject formData)
-            throws hudson.model.Descriptor.FormException {
-        JSONObject converted;
-        if (formData.containsKey(CONFIGURATION_SECTION_KEY)) {
-            JSONObject old = formData.getJSONObject(CONFIGURATION_SECTION_KEY);
-            formData.put(CONFIGURATION_SECTION_KEY, convertHierarchicalFormData(old));
-            converted = formData;
-        }
-        else {
-            converted = convertHierarchicalFormData(formData);
-        }
-        return super.newInstance(req, converted);
-    }
```

Comment:
By the way if you have time I would recommend adding a `configRoundTrip` test, which you would expect to pass in `master`, then fail when you only delete the `newInstance` override, but pass again when you switch to `inline="true"`.

You could also add a test using `DescribableHelper` to prove that Workflow configuration is round-tripped correctly only if you do both of these changes.